org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/DeltaProcessor.java

```diff
@@ -2022,7 +2023,9 @@ public void resetProjectCaches() {
 	 */
 	public void registerJavaModelDelta(IJavaElementDelta delta) {
 		if (JavaModelManager.isReadOnly()) {
-			throw new IllegalStateException("Its not allow to modify JavaModel during ReadOnly action. delta=" + delta); //$NON-NLS-1$
+			ILog.get().warn("JavaModel change during read only operation", new IllegalStateException( //$NON-NLS-1$
+					"JavaModel modified during 'read only' operation. Consider to report this warning to https://github.com/eclipse-jdt/eclipse.jdt.core/issues. delta=" //$NON-NLS-1$
+							+ delta));
```

Comment:
Do we tell the user to report errors to `github` anywhere else in logged messages? if not I would omit this. Especially on a warning.

Maybe this should be an error though, considering we are expecting trouble to follow after this log (I assume caching somehow will go wrong?).

Reply:
> Do we tell the user to report errors to github anywhere else in logged messages? if not I would omit this. Especially on a warning.

In other places we've suggested to report to bugzilla, but that time is over :-)
And yes, it would make sense if users would report these warnings, so we can fix inappropriate code.